micropub.php

```diff
@@ -625,17 +610,39 @@ public static function generate_post_content( $args ) {
 						$val['name']
 					);
 				}
+				$kindneeded = $prop;
 			}
 		}
 
 		if ( isset( $props['rsvp'] ) ) {
-			$lines[] = '<p>RSVPs <data class="p-rsvp" value="' . $props['rsvp'][0] .
+			$lines[]    = '<p>RSVPs <data class="p-rsvp" value="' . $props['rsvp'][0] .
 			  '">' . $props['rsvp'][0] . '</data>.</p>';
+			$kindneeded = 'rsvp';
+		}
+
+		$checkin = $props['checkin'][0];
+		if ( $checkin ) {
+			$name       = $checkin['properties']['name'][0];
+			$urls       = $checkin['properties']['url'];
+			$lines[]    = '<p>Checked into <a class="h-card p-location" href="' .
+				( $urls[1] ?: $urls[0] ) . '">' . $name . '</a>.</p>';
+			$kindneeded = 'checkin';
 		}
 
 		// event
 		if ( array( 'h-event' ) === static::$input['type'] ) {
-			$lines[] = static::generate_event( static::$input );
+			$lines[]    = static::generate_event( static::$input );
+			$kindneeded = 'event';
+		}
+		if ( isset( $kindmap[ $kindneeded ] ) ) {
+			$kindneeded = $kindmap[ $kindneeded ];
+		}
+
+		if ( ! $args['post_excerpt'] &&
+			  // Post Kinds: https://wordpress.org/plugins/indieweb-post-kinds/
```

Comment:
minor nit: this needs a bit more elaboration, e.g. that Post Kinds sets `kind_termslist` to an array of string kinds that the user has enabled.

Reply:
I have done as you ask.